src/main/webapp/scripts/keywords.js

```diff
@@ -1,6 +1,47 @@
 const BLOB_URL_PROMISE = loadTemplate('/user-image');
 
 $().ready(function() {
+  fetch('/login').then(response => response.json()).then((logURL) => {
+    $('#log-btn > a').attr('href', logURL[0]);
+    if (logURL.length == 2) {  // User is not loggeed in.
+      $('#form-title > p').text('You are not logged in.');
+      $('#log-btn > a').text("Login");
+      $('#keyword-form, #input-select, #location-input, #previous-keywords').addClass('hide');
+    } else {
+      $('#form-title > p').text('Choose Type of Input');
+      $('#log-btn > a').text('Logout');
+      $('#input-select, #location-input, #previous-keywords').removeClass('hide');
+    }
+    $('#log-btn').removeClass('hide');
+  });
+
+  fetch('/all-keywords').then(response => response.json()).then((keywordMap) => {
```

Comment:
[Optional] Same as above. Preload the object and use async-await keywords instead of chaining promises.

Reply:
I will look into this.